tests/unit/uart_test.cpp

```diff
@@ -107,13 +120,25 @@ TEST_F(UartTest, InitFailures) {
     EXPECT_EQ(W_IO_ERROR, status);
 
     // Test RX Event callback registration failure
-    xQueueCreate_fake.return_val = (QueueHandle_t)1;
     HAL_UART_RegisterCallback_fake.return_val = HAL_OK;
+    xQueueCreate_fake.return_val = (QueueHandle_t)1;
     HAL_UART_RegisterRxEventCallback_fake.return_val = HAL_ERROR;
     status = uart_init(TEST_CHANNEL, &huart, timeout);
     EXPECT_EQ(W_FAILURE, status);
 }
 
+// Test uart_write with valid message
+TEST_F(UartTest, WriteSuccess) {
+    uint8_t buffer[UART_MAX_LEN];
+    uint8_t length;
+    xSemaphoreTake_fake.return_val = pdTRUE;
+    xSemaphoreGive_fake.return_val = pdTRUE;
+    w_status_t status = uart_write(TEST_CHANNEL, buffer, &length, timeout);
+    EXPECT_EQ(W_SUCCESS, status);
```

Comment:
Maybe we could also check that the mutex and semaphore behaviour is correct?
```suggestion
    EXPECT_EQ(W_SUCCESS, status);
    EXPECT_EQ(2, xSemaphoreTake_fake.call_count);
    EXPECT_EQ(1, xSemaphoreGive_fake.call_count);
```